template/src/VideoSample/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, {useCallback, useEffect, useMemo} from 'react';
+import * as RN from 'react-native';
+import {Types, Fonts, Theme} from '@sleeperhq/mini-core';
+import {FlashList} from '@shopify/flash-list';
+import {Topic} from '@sleeperhq/mini-core/declarations/types';
+import VideoRow from './VideoRow';
+
+type VideoMiniProps = {
+  context: Types.Context;
+};
+
+const PAGE_LIMIT = 20;
+const EMPTY_OBJECT = {};
+
+const Videos = (props: VideoMiniProps) => {
+  const {context} = props;
+
+  const [page, setPage] = React.useState(0);
+  const [isRefreshing, setIsRefreshing] = React.useState(false);
+  const [paginatedVideosMap, setPaginatedVideoMap] =
```

Comment:
If it's a non-standard way, I feel a bit hesitant to use it in our sample code. Can we easily test this impact?

Reply:
Hmm. After looking at the code I do feel like it's a little bit odd to include it here in this project. We have a couple of options - 

1) Rather than having `PodcastSample` and `VideoSample` live in the same directory level as the other samples, we move it one level deeper. The shared helpers file then would live within that new folder. So, for example:
```
- src
    - Mini
    - FetchSample
    - ActionSample
    - live
          - Podcasts
          - Videos
          - shared
                - helpers
```
In app.json, we would then update the samples list to:
```
"samples": ["Mini", "FetchSample", "ActionSample", "live/Podcasts", "live/Videos"],
```

2) Add `useMergeState` to sleeper mini core as a helper method. It's useful enough and a big enough performance improvement that it might be worth it. But it's also an artifact of being on an older react native version that doesn't handle state update batching automatically.

Reply:
@sleeper-luke how do we test that #2 hasn't been corrected in our current version of react native?

Reply:
I think #1 is totally fine, will #2 be hard to implement? Just wondering what I should do for the purposes of this PR. Just hold off for a little? Is that something I can add and expose through context?

Reply:
@sleeper-jasonl you can just put a print statement in render/useEffect. you'll get multiple logs each time we trigger the refresh with the sequence of three `setState()` calls.

And yeah, @taylorsleeper, let's just go with #1 for now. Hide the helpers file a bit so it's more obviously internal to just our own minis and not meant to be used for other projects yet.